### PR TITLE
Added multiple controllers support.  Fixes #5

### LINKED ISSUE #5: Multiple controllers
Add support for multiple Controller= as a stopgap for ACLs.

### PR DESCRIPTION
**Pull request by SuperSpyTX**
See #5 

This works with having comma separated values for the Controller= string.  I'm not sure if this is something that would be ideal or if you wanted something different.  But it does work.

### CODE REVIEW

File: team_config.go [hunk 1]
```diff
@@ -43,9 +43,20 @@ func LoadTeamConfig(sec *ini.Section) *TeamConfig {
...
+	if strings.ContainsAny(controllerKey, ",") {
+		var split = strings.Split(controllerKey, ",")
+		c.Controller = make([]slack.UserID, len(split))
+		for uid := range c.Controller {
+			c.Controller[uid] = slack.UserID(split[uid])
+		}
+	} else {
+		c.Controller = []slack.UserID{slack.UserID(controllerKey)}
+	}
+
```
Comment (SuperSpyTX) [inline team_config.go]: How do you feel about this extra whitespace?

File: team_config.go [hunk 2]
```diff
@@ -26,7 +26,7 @@ type TeamConfig struct {
 	LogChannel      slack.ChannelID
 	HTTPListen      string
 	HTTPURL         string
-	Controller      slack.UserID
+	Controller      []slack.UserID
```
Comment (riking) [inline team_config.go]: plural

File: slack/controller/channel_info.go [hunk 3]
```diff
@@ -101,8 +101,10 @@ func (t *Team) UserName(user slack.UserID) string {
 }
 
 func (t *Team) UserLevel(user slack.UserID) marvin.AccessLevel {
-	if user == t.TeamConfig().Controller {
-		return marvin.AccessLevelController
+	for id := range t.TeamConfig().Controller {
```
Comment (riking) [inline slack/controller/channel_info.go]: move into method TeamConfig.IsController(string)bool

File: team_config.go [hunk 4]
```diff
@@ -43,9 +43,20 @@ func LoadTeamConfig(sec *ini.Section) *TeamConfig {
 	c.HTTPListen = sec.Key("HTTPListen").String()
 	c.HTTPURL = sec.Key("HTTPURL").String()
 	c.LogChannel = slack.ChannelID(sec.Key("LogChannel").String())
-	c.Controller = slack.UserID(sec.Key("Controller").String())
 	c.IsDevelopment, _ = sec.Key("IsDevelopment").Bool()
 
+	var controllerKey = sec.Key("Controller").String()
+	if strings.ContainsAny(controllerKey, ",") {
+		var split = strings.Split(controllerKey, ",")
+		c.Controller = make([]slack.UserID, len(split))
```
Comment (riking) [inline team_config.go]: I think you can just cast the []string to a []slack.UserID?

File: team_config.go (hunk 1 above)
Comment (riking) [reply team_config.go]: Run the formatter :wink:

File: team_config.go [hunk 5]
```diff
@@ -43,9 +43,20 @@ func LoadTeamConfig(sec *ini.Section) *TeamConfig {
...
+		for uid := range c.Controller {
+			c.Controller[uid] = slack.UserID(split[uid])
+		}
+	} else {
```
Comment (riking) [inline team_config.go]: No need for the branch, `strings.Split` will just return the single item.

File: team_config.go (hunk 4 above)
Comment (SuperSpyTX) [reply team_config.go]: screen shot 2017-08-29 at 4 31 28 pm

File: team_config.go [hunk 6]
```diff
@@ -61,6 +67,15 @@ func LoadTeamConfig(sec *ini.Section) *TeamConfig {
 	return c
 }
 
+func (t *TeamConfig) IsController(user string) bool {
```
Comment (riking) [inline team_config.go]: the function can take a slack.UserID to avoid the string casts at the callsite and the function

Comment (SuperSpyTX) [reply team_config.go]: That was what I was going to do initially, but I didn't know if you specifically needed it to be a string or not.

**Pull request merged**